File: Data Structures & Algorithms/non-cyclical-number/submission-12.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def helper(num):
            c = 0
            for i in str(num):
                c += int(i)*int(i)
            return c
        seen = set()
        while True:
            n = helper(n)
            if n == 1:
                return True
            if n in seen:
                return False
            seen.add(n)
            
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-12.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        for (int i = 0; i < nums.size(); i++){
            if (seen.find(nums[i]) == seen.end()) {
                seen[nums[i]] = i;
            }
        for (int i = 0; i<nums.size(); i++){
            int compliment = target - nums[i];
            if (seen.find(compliment) != seen.end() && seen[compliment] != i) {
                return {seen[compliment],i};
            }
        }
        }
    }
};
